experiments/modern/src/maki-interpreter/virtualMachine.js

```diff
@@ -32,9 +32,11 @@ async function interpret(start, program, stack = [], { logger = null }) {
     stack.push(operator(bValue, aValue));
   }
 
-  // Run all the commands that are safe to run. Increment this number to find
-  // the next bug.
-  const maxOffset = Math.max(...Object.keys(offsetToCommand));
+  function jumpToOffset(offset) {
+    const nextCommandIndex = offsetToCommand[offset];
+    return nextCommandIndex - 1;
+  }
```

Comment:
Nice! Eventually I would like to move this into the parser, but having it in one place here in the VM is a huge first step to being able to rip it out.

Reply:
oh hadn't even thought about how we could just do that in the parser once and be done, sounds good!

Reply:
It's going to be a little annoying, since we will need to do it _after_ all the command parsing is done. That means we will need to do a second pass which knows which opcodes have arguments which are command offsets.